pkg/apis/stork/v1alpha1/schedulepolicy.go

```diff
@@ -0,0 +1,182 @@
+
+// SchedulePolicy represents a policy for executing actions on a schedule
+type SchedulePolicy struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	// Policy
+	Policy SchedulePolicyItem `json:"policy"`
+}
+
+// SchedulePolicyType is the type of schedule policy
+type SchedulePolicyType string
+
+const (
+	// SchedulePolicyTypeInvalid is an invalid schedule policy
+	SchedulePolicyTypeInvalid SchedulePolicyType = "Invalid"
+	// SchedulePolicyTypeInterval is the type for an interval schedule policy
+	SchedulePolicyTypeInterval SchedulePolicyType = "Interval"
+	// SchedulePolicyTypeDaily is the type for a daily schedule policy
+	SchedulePolicyTypeDaily SchedulePolicyType = "Daily"
+	// SchedulePolicyTypeWeekly is the type for a weekly schedule policy
+	SchedulePolicyTypeWeekly SchedulePolicyType = "Weekly"
+	// SchedulePolicyTypeMonthly is the type for a monthly schedule policy
+	SchedulePolicyTypeMonthly SchedulePolicyType = "Monthly"
+)
+
+// GetValidSchedulePolicyTypes returns the valid types of schedule policies that
+// can be configured
+func GetValidSchedulePolicyTypes() []SchedulePolicyType {
+	return []SchedulePolicyType{SchedulePolicyTypeInterval, SchedulePolicyTypeDaily, SchedulePolicyTypeWeekly, SchedulePolicyTypeMonthly}
+}
+
+// Days is a map of valid Day strings
+var Days = map[string]time.Weekday{
+	"Sunday":    time.Sunday,
+	"Sun":       time.Sunday,
+	"Monday":    time.Monday,
+	"Mon":       time.Monday,
+	"Tuesday":   time.Tuesday,
+	"Tue":       time.Tuesday,
+	"Wednesday": time.Wednesday,
+	"Wed":       time.Wednesday,
+	"Thursday":  time.Thursday,
+	"Thu":       time.Thursday,
+	"Thur":      time.Thursday,
+	"Thurs":     time.Thursday,
+	"Friday":    time.Friday,
+	"Fri":       time.Friday,
+	"Saturday":  time.Saturday,
+	"Sat":       time.Saturday,
+}
+
+// SchedulePolicyItem represents the schedule for executing an action
+type SchedulePolicyItem struct {
+	// Interval policy that will be triggered at the specified interval
+	Interval *IntervalPolicy `json:"interval"`
+	// Daily policy that will be triggered daily at a specified time
+	Daily *DailyPolicy `json:"daily"`
+	// Weekly policy that will be triggered on the specified day of the week at
+	// the specified time
+	Weekly *WeeklyPolicy `json:"weekly"`
+	// Monthly policy that will be triggered on the specified date of the month
+	// at the specified time
+	Monthly *MonthlyPolicy `json:"monthly"`
+}
+
+// IntervalPolicy contains the interval at which an action should be triggered
+type IntervalPolicy struct {
+	IntervalMinutes int `json:"intervalMinutes"`
+}
+
+// Validate validates an IntervalPolicy
+func (i *IntervalPolicy) Validate() error {
+	if i.IntervalMinutes < 1 {
+		return fmt.Errorf("Invalid intervalMinutes (%v) in Interval policy", i.IntervalMinutes)
+	}
+	return nil
+}
+
+// DailyPolicy contains the time in the day where an action should be executed
+type DailyPolicy struct {
+	Time string `json:"time"`
```

Comment:
What format does `Time` follow? Please add that to the comment.

Reply:
Done. Added comments for the other members too.